### ISSUE TITLE
[Bug Report] `DtypeObservation` not working for `float32` -> `float64` in vector envs

### ISSUE THREAD

**Issue by jjshoots** [Farama-Foundation/Gymnasium#705]
### Describe the bug

Not sure if this is intended, but the `gymnasium.wrappers.vector.DtypeObservation` doesn't work for converting a `float32` env into a `float64` one.

### Code example

```python
>>> import numpy as np
>>> import gymnasium as gym
>>> envs = gym.make_vec("CartPole-v1", num_envs=3)
>>> obs, info = envs.reset(seed=123)
>>> envs.close()
>>> obs.dtype
dtype('float32')
>>> envs = gym.make_vec("CartPole-v1", num_envs=3)
>>> envs = DtypeObservation(envs, dtype=np.float64)
>>> obs, info = envs.reset(seed=123)
>>> envs.close()
>>> obs.dtype
dtype('float32')
```

### System info

Gymnasium v1.0.0

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/Gymnasium/issues) in the repo

**Comment by pseudo-rnd-thoughts** [comment]
https://github.com/Farama-Foundation/Gymnasium/blob/74e6e13b78188068fdf5a595fb409926dec48fa8/gymnasium/spaces/box.py#L277 is the cause.
I can't remember why I have this commented out, I suspect that there is an issue with something else, I've made #707 to experiment

**Comment by pseudo-rnd-thoughts** [comment]
Fixed by #707
